sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ListStateImplWithTTL.scala

```diff
@@ -137,6 +137,7 @@ class ListStateImplWithTTL[S](
   /** Remove this state. */
   override def clear(): Unit = {
     store.remove(stateTypesEncoder.encodeGroupingKey(), stateName)
+    clearTTLState()
```

Comment:
This was a miss on the `ListState` PR ?

Reply:
Yes, it was. @sahnib realized that when we clear state, we can also clear the associated ttlState